src/Builder/Pipeline.php

```diff
@@ -36,7 +39,9 @@ public function __construct(StageInterface|Pipeline ...$stagesOrPipelines)
         $stages = [];
 
         foreach ($stagesOrPipelines as $stageOrPipeline) {
-            if ($stageOrPipeline instanceof Pipeline) {
+            if (is_array($stageOrPipeline) && array_is_list($stageOrPipeline)) {
```

Comment:
I noticed you don't check whether each item in the list is a `StageInterface` instance. In theory, this should be fine as `PipelineEncoder` uses `encodeIfSupported` to encode each stage. Anything the encoder can't handle will be sent to the server as is, potentially causing an execution error later. This would allow users to inject raw pipeline stages (e.g. if they want to use something not supported by their builder version) as long as they know what they are doing.